CRM/Core/BAO/Email.php

```diff
@@ -252,7 +252,7 @@ public static function holdEmail(&$email) {
           $email->reset_date = 'null';
         }
       }
-      elseif ($email->on_hold == 'null') {
+      elseif ($email->on_hold == 'null' || empty($email->on_hold)) {
```

Comment:
I think this should be $email->on_hold === 0 (since it might be an update that is leaving that empty for 'no change')

Unfortunately it would also need a unit test & I'm guessing that since it's been around a long time you are probably not keen to jump on that.

I'm going to close it & keep tracking through JIRA since it's not mergeable as is. If you do wish to pick it up & work on a test please open a new PR (which will make it much more visible too)